soapui/src/main/java/com/eviware/soapui/impl/wsdl/submit/transports/http/HttpClientRequestTransport.java

```diff
@@ -313,9 +318,21 @@ private boolean isRedirectResponse(int statusCode) {
         return false;
     }
 
-    private ExtendedGetMethod followRedirects(HttpClient httpClient, int redirectCount, ExtendedHttpMethod httpMethod,
-                                              org.apache.http.HttpResponse httpResponse, HttpContext httpContext) throws Exception {
-        ExtendedGetMethod getMethod = new ExtendedGetMethod();
+    private ExtendedHttpMethod followRedirects(HttpClient httpClient, int redirectCount, ExtendedHttpMethod httpMethod,
+                                              org.apache.http.HttpResponse httpResponse, HttpContext httpContext, SubmitContext submitContext) throws Exception {
+        EntityEnclosingRequestWrapper httpRequestWrapper = (EntityEnclosingRequestWrapper)httpContext.getAttribute(org.apache.http.protocol.ExecutionContext.HTTP_REQUEST);
+		int statusCode = httpResponse.getStatusLine().getStatusCode();
+
+		ExtendedHttpMethod getMethod;
+		if((statusCode != 303) && (httpRequestWrapper != null) && (httpRequestWrapper.getMethod().equals(RestRequestInterface.HttpMethod.POST.toString())))

```

Comment:
I have two problems with this conditional.
1. You should use HttpServletResponse.SC_SEE_OTHER rather than the "magic number" 303.
2. The parentheses around the different conditional statements seem unnecessary to me and are not consistent with the code style in the rest of SoapUI. 

I agree that the conditional is too complex though, so I recommend extracting a method with a good name that explains what you're checking.